### PR TITLE
Love: Allow custom title per entity in entities card

### PR DESCRIPTION
**Pull request by jeradM**
Modified all `state-card-*` entities to allow passing of lovelace entity config. 

Addresses https://github.com/home-assistant/ui-schema/issues/36

### CODE REVIEW

**Comment by jeradM** [comment]
@c727 are you able to restart the travis build? It failed on downloading node

**Comment by arsaboo** [comment]
@jeradM You can close and reopen the PR to re-trigger the build.

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,6 @@ dist
 # Secrets
 .lokalise_token
 yarn-error.log
+
+# JetBrains
```
Comment (c727) [inline .gitignore]: please exclude your addiotions here

File: src/components/entity/state-info.js [hunk 2]
```diff
@@ -82,12 +82,13 @@ class StateInfo extends PolymerElement {
       },
       hass: Object,
       stateObj: Object,
-      inDialog: Boolean
+      inDialog: Boolean,
+      entityConfig: Object
```
Comment (c727) [inline src/components/entity/state-info.js]: can we name it `config` for consistency?

File: src/components/entity/state-info.js [hunk 3]
```diff
@@ -82,12 +82,13 @@ class StateInfo extends PolymerElement {
       },
       hass: Object,
       stateObj: Object,
-      inDialog: Boolean
+      inDialog: Boolean,
+      config: Object
```
Comment (balloob) [inline src/components/entity/state-info.js]: This is wrong. These are components that are not part of lovelace and thus should not be made aware of lovelace config

Comment (balloob) [reply src/components/entity/state-info.js]: We should just have a title property that will be used if set.

Although I think name is better here. We override the state name that is being calculated after all.

File: src/state-summary/state-card-climate.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ class StateCardClimate extends PolymerElement {
 
   static get stateInfoTemplate() {
     return html`
-    <state-info hass="[[hass]]" state-obj="[[stateObj]]" in-dialog="[[inDialog]]"></state-info>
+    <state-info hass="[[hass]]" state-obj="[[stateObj]]" in-dialog="[[inDialog]]" config="[[config]]"></state-info>
```
Comment (balloob) [inline src/state-summary/state-card-climate.js]: Let's introduce new lines between attributes and not create such long lines. Also, we should not make these components config aware.

File: src/state-summary/state-card-climate.js [hunk 5]
```diff
@@ -42,6 +42,7 @@ class StateCardClimate extends PolymerElement {
         type: Boolean,
         value: false,
       },
+      config: Object
```
Comment (balloob) [inline src/state-summary/state-card-climate.js]: This should not be called config but maybe `overrideName` ?

File: src/components/entity/state-info.js (hunk 3 above)
Comment (c727) [reply src/components/entity/state-info.js]: What do you think about creating a new stateObj with our custom name as friendly_name before we pass it to the state-cards?

**Review by balloob — APPROVED** [review]

**Pull request merged**